### ISSUE TITLE
Howto create a fragment with nested fragments from a json string

### ISSUE THREAD

**Issue by cyphercrack** [lytics/ember-data.model-fragments#105]
Hi,
is there an easy way to create new fragments with nested fragments from a json string, without parsing the json string by myself an call createFragment for every nested fragment? 
When the ember data object is initially created from the REST call the fragments are created automatically. How can i use this mechanism?

Best regards
Michael

**Comment by slindberg** [comment]
TL;DR is no, not right now.

I assume that you already have normalized data for the fragment, and want to do something like:

```javascript
someModel.set('things', {
  foo: 'bar',
  baz: {
    quuz: 'bang'
  }
});

Where `baz` is a nested fragment (if this is not what you're trying to do, please clarify). Unfortunately setting fragment properties with normalized JSON is not currently supported, but I've heard enough requests for it that it's on my TODO list.

**Comment by cyphercrack** [comment]
Yes, and further, quuz can also be a nested fragment.

**Comment by christophersansone** [comment]
@cyphercrack Does this do the job for you?

```javascript
var data = { foo: 'bar', child: { bar: 'baz' } };
var fragment = MyModel.createFragment();
fragment.setupData(data);
```

@slindberg is correct that the JSON must be denormalized as if it came from the server.

**Comment by slindberg** [comment]
@cyphercrack Be careful using `setupData`; after thinking about this a little more, I think this won't copy sub fragments, but instead overwrite their data. Also keep in mind that it's a private method and subject to change.

**Comment by slindberg** [comment]
Support added in #144, released in v0.4.2
